src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,85 +1,1000 @@
+                        } else {
+                            BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN) = BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT);
+                            C_SLOT_EQUIP(0, EQUIP_SLOT_C_DOWN) = C_SLOT_EQUIP(0, EQUIP_SLOT_C_RIGHT);
+                            Interface_LoadItemIcon(play, EQUIP_SLOT_C_DOWN);
+                        }
+                    } else {
+                        BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN) = ITEM_NONE;
+                        C_SLOT_EQUIP(0, EQUIP_SLOT_C_DOWN) = SLOT_NONE;
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081B6EC.s")
+                // Special case for magic arrows
+                if ((pauseCtx->equipTargetItem >= 0xB5) && (pauseCtx->equipTargetItem < 0xB8)) {
+                    if ((BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT) == ITEM_BOW) ||
+                        ((BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT) >= ITEM_BOW_ARROW_FIRE) &&
+                         (BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT) <= ITEM_BOW_ARROW_LIGHT))) {
+                        pauseCtx->equipTargetItem -= 0xB5 - ITEM_BOW_ARROW_FIRE;
+                        pauseCtx->equipTargetSlot = SLOT_BOW;
+                    }
+                } else if (pauseCtx->equipTargetItem == ITEM_BOW) {
+                    if ((BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_LEFT) >= ITEM_BOW_ARROW_FIRE) &&
+                        (BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_LEFT) <= ITEM_BOW_ARROW_LIGHT)) {
+                        BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_LEFT) = BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT);
+                        Interface_LoadItemIcon(play, EQUIP_SLOT_C_LEFT);
+                    } else if ((BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN) >= ITEM_BOW_ARROW_FIRE) &&
+                               (BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN) <= ITEM_BOW_ARROW_LIGHT)) {
+                        BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN) = BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT);
+                        Interface_LoadItemIcon(play, EQUIP_SLOT_C_DOWN);
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081BCA8.s")
+                // Equip item on CRight
+                BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT) = pauseCtx->equipTargetItem;
+                C_SLOT_EQUIP(0, EQUIP_SLOT_C_RIGHT) = pauseCtx->equipTargetSlot;
+                Interface_LoadItemIconImpl(play, EQUIP_SLOT_C_RIGHT);
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081C684.s")
+            // Reset params
+            pauseCtx->unk_200 = 0;
+            sEquipAnimTimer = 10;
+            pauseCtx->equipAnimScale = 320;
+            pauseCtx->equipAnimShrinkRate = 40;
+        }
+    } else {
+        sEquipMagicArrowSlotHoldTimer--;
+        if (sEquipMagicArrowSlotHoldTimer == 0) {
+            pauseCtx->equipAnimAlpha = 255;
+        }
+    }
+}
```

Comment:
I feel like a lot of the early returns in this function might be replaceable by elses, but at the same time it's clear that the file in general loves early returns so I don't think it's worth worrying about.

Reply:
In a lot of cases, you definitely can replace early returns with early else's. But the nested scope goes many layers deep with these large functions dense with if-statements, and I find it vastly improves readability for these more larger and complex functions. I know it does go against the style of preferring "else" over early returns, but there's been a general exception to this in both OoT and MM for larger functions with many layer-deep nested ifs